soul-admin/src/test/java/org/dromara/soul/admin/service/SoulClientRegisterServiceTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dromara.soul.admin.service;
+
+import org.dromara.soul.admin.dto.MetaDataDTO;
+import org.dromara.soul.admin.dto.SpringCloudRegisterDTO;
+import org.dromara.soul.admin.dto.SpringMvcRegisterDTO;
+import org.dromara.soul.admin.service.impl.SoulClientRegisterServiceImpl;
+import org.dromara.soul.admin.utils.SoulResultMessage;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+/**
+ * Test cases for SoulClientRegisterService.
+ *
+ * @author sakiila
+ */
+@RunWith(MockitoJUnitRunner.class)
+public final class SoulClientRegisterServiceTest {
+
+    @Mock
+    private SoulClientRegisterServiceImpl soulClientRegisterService;
+
+    @Test
+    public void testRegisterSpringMvc() {
+        SpringMvcRegisterDTO dto = buildSpringMvcRegisterDTO();
+        given(soulClientRegisterService.registerSpringMvc(any())).willReturn(SoulResultMessage.SUCCESS);
```

Comment:
You can also mock metaDataMapper and eventPublisher to test more scenarios. Thanks.